docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+
+And then create a type called ``SomeType`` without any properties:
+
+.. code-block:: edgeql-repl
+
+    db> create type SomeType;
+    OK: CREATE TYPE
+
+Your schema now contains this type, as you can see by typing ``describe 
+schema`` or ``describe schema as sdl``:
+
+.. code-block::
+
+    {
+    'module default {
+        type SomeType;
+        type User {
+            property name: std::str;
+        };
+    };',
+    }
+
+Great! This type is now inside your schema and you can do whatever 
+you like with it.
+
+But this has also ruined the migration flow. Watch what happens when 
+you try to apply the change:
+
+.. code-block:: edgeql-repl
+
+    db> \migration create
+    Error executing command: Database must be updated to 
+    the last migration on the filesystem for 
+    `migration create`. Run:
+    edgedb migrate
+
+    db> \migrate
+    Error executing command: database applied migration 
+    history is ahead of migration history in 
+    "myproject/dbschema/migrations" by 1 revision
+
+Sneakily adding ``SomeType`` into your schema to match won't work 
+either. The problem is that there *is* a migration already present, 
+it just doesn't exist inside your ``/migrations`` folder. You can 
+see it with the following query:
+
+.. code-block:: edgeql-repl
+
+    db> select schema::Migration {*} 
+    ...  filter 
+    ...  .generated_by = schema::MigrationGeneratedBy.DDLStatement;
+    {
+    schema::Migration {
+        id: 3882894a-8bb7-11ee-b009-ad814ec6a5f5,
+        name: 'm1s6oniru3zqepiaxeljt7vcgyynxuwh4ki3zdfr4hfavjozsndfua',
+        internal: false,
+        builtin: false,
+        computed_fields: [],
+        script: 'SET generated_by := 
+            (schema::MigrationGeneratedBy.DDLStatement);
+    CREATE TYPE SomeType;',
+        message: {},
+        generated_by: DDLStatement,
+    },
+    }
+
+Fortunately, the fix is not too hard: we can use a command called 
+``edgedb migration extract``. This command will retrieve the migration(s) 
+created using DDL and assign each of them a proper file name and hash 
+inside the ``/migrations`` folder, effectively giving them a proper 
+position inside the migration flow.
+
+Note that at this point your ``.esdl`` schema will still not match 
+the database schema, so if you were to type ``edgedb migration create`` 
+the CLI would then ask you if you want to drop the type that you just 
+created - because it doesn't exist inside there. So be sure to change 
+your schema to match the schema inside the database that you have 
+manually changed via DDL. If in doubt, use ``describe schema as sdl`` 
+to compare or use ``edgedb migration create`` and check the output. 
+If the CLI is asking you if you want to drop a type, that means that 
+you forgot to add it to the schema inside your ``.esdl`` file(s).
+
+
+Multiple migrations to keep data 
+================================
+
+Let's say that we have a simple ``User`` type with a ``friends`` link 
```

Comment:
We should either tell readers _not_ to follow along or give them instructions how they should. Do they migrate what they have, or start a new project?

Reply:
True that it should be more inviting: https://github.com/edgedb/edgedb/pull/6656/commits/f4cea9899aa0da77125e002e770d8a9f3bd15f25